File: src/app/core/models/response/Order/OrderResponse.ts
import { OrderStatus } from "src/app/utils/OrderStatus";
import { PaymentMethod } from "src/app/utils/PaymentMethod";
import { PaymentStatus } from "src/app/utils/PaymentStatus";
import { OrderItemResponse } from "./OrderItemResponse";

export interface OrderResponse {
  id: number;
  orderNumber: string;
  userId: number;
  username?: string;
  totalAmount: number;
  shippingAddress: string;
  phoneNumber: string;
  receiverName: string;
  status: OrderStatus;
  paymentStatus: PaymentStatus;
  paymentMethod: PaymentMethod;
  note?: string;
  items: OrderItemResponse[];
  createdAt: string;
  updatedAt: string;
}

File: demo/src/main/java/com/training/demo/dto/response/Order/OrderResponse.java
package com.training.demo.dto.response.Order;

import com.training.demo.utils.enums.OrderStatus;
import com.training.demo.utils.enums.PaymentMethod;
import com.training.demo.utils.enums.PaymentStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderResponse {
    private Long id;
    private String orderNumber;
    private Long userId;
    private String username;
    private BigDecimal totalAmount;
    private String shippingAddress;
    private String phoneNumber;
    private String receiverName;
    private OrderStatus status;
    private PaymentStatus paymentStatus;
    private PaymentMethod paymentMethod;
    private String note;
    private List<OrderItemResponse> items;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
